libbeat/processors/script/javascript/module/processor/processor_test.go

```diff
@@ -97,299 +74,41 @@ function process(evt) {
-    ],
-});
-
-function process(evt) {
-	copy.Run(evt);
-}
-`
-
-	logp.TestingSetup()
-	p, err := javascript.NewFromConfig(javascript.Config{Source: script}, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	evt, err := p.Run(testEvent())
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	_, err = evt.GetValue("log.original")
-	assert.NoError(t, err)
-}
-
-func TestNewProcessorDecodeJSONFields(t *testing.T) {
-	const script = `
-var processor = require('processor');
-
-var decodeJSON = new processor.DecodeJSONFields({
-    fields: ["message"],
-    target: "",
-});
-
-function process(evt) {
-	decodeJSON.Run(evt);
-}
-`
-
-	logp.TestingSetup()
-	p, err := javascript.NewFromConfig(javascript.Config{Source: script}, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	evt := testEvent()
-	evt.PutValue("message", `{"hello": "world"}`)
 
-	_, err = p.Run(evt)
-	if err != nil {
-		t.Fatal(err)
+	switch ss := s.(type) {
+	case string:
+		assert.Equal(t, ss, "new_value")
+	default:
+		t.Fatal("unexpected type")
 	}
-
-	v, _ := evt.GetValue("hello")
-	assert.Equal(t, "world", v)
 }
 
-func TestNewProcessorDissect(t *testing.T) {
-	const script = `
-var processor = require('processor');
-
-var chopLog = new processor.Dissect({
-    tokenizer: "key=%{key}",
-    field: "message",
-});
-
-function process(evt) {
-    chopLog.Run(evt);
+type mockProcessor struct {
+	fields common.MapStr
 }
-`
-
-	logp.TestingSetup()
-	p, err := javascript.NewFromConfig(javascript.Config{Source: script}, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
 
-	evt, err := p.Run(testEvent())
+func newMock(c *common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields common.MapStr `config:"fields" validate:"required"`
+	}{}
+	err := c.Unpack(&config)
 	if err != nil {
-		t.Fatal(err)
+		return nil, fmt.Errorf("fail to unpack the mock processor configuration: %s", err)
 	}
 
-	key, _ := evt.GetValue("dissect.key")
-	assert.Equal(t, "hello", key)
+	return &mockProcessor{
+		fields: config.Fields,
+	}, nil
 }
 
-func TestNewProcessorDNS(t *testing.T) {
-	if runtime.GOOS == "windows" {
-		t.Skip("Windows requires explicit DNS server configuration")
-	}
-
-	const script = `
-var processor = require('processor');
-
-var dns = new processor.DNS({
-    type: "reverse",
-    fields: {
-        "source.ip": "source.domain",
-        "destination.ip": "destination.domain"
-    },
-    tag_on_failure: ["_dns_reverse_lookup_failed"],
-});
-
-function process(evt) {
-	dns.Run(evt);
-    if (evt.Get().tags[0] !== "_dns_reverse_lookup_failed") {
-        throw "missing tag";
-    }
+func (m *mockProcessor) Run(event *beat.Event) (*beat.Event, error) {
+	event.Fields.DeepUpdate(m.fields)
+	return event, nil
 }
-`
 
-	logp.TestingSetup()
-	p, err := javascript.NewFromConfig(javascript.Config{Source: script}, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	_, err = p.Run(testEvent())
-	if err != nil {
-		t.Fatal(err)
-	}
-}
-
-func TestNewRename(t *testing.T) {
```

Comment:
Just curious, is there any special reason we are moving these tests to python?

Reply:
Mostly because registering the modules are done in a different package. By testing the external processors here, all imported processors became too tightly coupled with this one IMO.  If anything changes in those "external" processors, there is no need to do any follow-ups in these unit tests anymore.  Or if these tests break, we know that the issue is with the script processor, not with other 10-15 processors which are imported. Thus, tracking down issues in the code is easier.